pkg/activator/net/throttler_test.go

```diff
@@ -0,0 +1,201 @@
+package activator
+
+import (
+	"context"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/google/go-cmp/cmp"
+	"k8s.io/apimachinery/pkg/types"
+
+	. "knative.dev/pkg/logging/testing"
+	"knative.dev/serving/pkg/apis/serving/v1alpha1"
+	servingfake "knative.dev/serving/pkg/client/clientset/versioned/fake"
+	servinginformers "knative.dev/serving/pkg/client/informers/externalversions"
+	"knative.dev/serving/pkg/queue"
+)
+
+const defaultMaxConcurrency = 1000
+
+type tryResult struct {
+	Dest      string
+	Ret       error
+	ErrString string
+}
+
+func TestThrottler(t *testing.T) {
+	for _, tc := range []struct {
+		name             string
+		revisions        []*v1alpha1.Revision
+		initUpdates      []*RevisionDestsUpdate
+		expectTryResults []tryResult
+		trys             []types.NamespacedName
+	}{{
+		name: "single healthy podIP",
+		revisions: []*v1alpha1.Revision{
+			revision(types.NamespacedName{"test-namespace", "test-revision"}),
+		},
+		initUpdates: []*RevisionDestsUpdate{{
+			Rev:               types.NamespacedName{"test-namespace", "test-revision"},
+			Dests:             []string{"128.0.0.1:1234"},
+			ReadyAddressCount: 1,
+		}},
+		trys: []types.NamespacedName{
+			{Namespace: "test-namespace", Name: "test-revision"},
+		},
+		expectTryResults: []tryResult{
+			{Dest: "128.0.0.1:1234"},
+		},
+	}, {
+		name: "single healthy clusterIP",
+		revisions: []*v1alpha1.Revision{
+			revision(types.NamespacedName{"test-namespace", "test-revision"}),
+		},
+		initUpdates: []*RevisionDestsUpdate{{
+			Rev:               types.NamespacedName{"test-namespace", "test-revision"},
+			ClusterIPDest:     "129.0.0.1:1234",
+			ReadyAddressCount: 1,
+		}},
+		trys: []types.NamespacedName{
+			{Namespace: "test-namespace", Name: "test-revision"},
+		},
+		expectTryResults: []tryResult{
+			{Dest: "129.0.0.1:1234"},
+		},
+	}, {
+		name: "spread podIP load",
+		revisions: []*v1alpha1.Revision{
+			revision(types.NamespacedName{"test-namespace", "test-revision"}),
+		},
+		initUpdates: []*RevisionDestsUpdate{{
+			Rev:               types.NamespacedName{"test-namespace", "test-revision"},
+			Dests:             []string{"128.0.0.1:1234", "128.0.0.2:1234"},
+			ReadyAddressCount: 2,
+		}},
+		trys: []types.NamespacedName{
+			{Namespace: "test-namespace", Name: "test-revision"},
+			{Namespace: "test-namespace", Name: "test-revision"},
+		},
+		expectTryResults: []tryResult{
+			{Dest: "128.0.0.1:1234"},
+			{Dest: "128.0.0.2:1234"},
+		},
+	}, {
+		name: "multiple clusterip requests after podip",
```

Comment:
I'm not quite understanding this test case. Looking at `func (rw *revisionWatcher) checkDests()`, there doesn't seem to be a use case for both `ClusterIPDest` and `Dests` to be set in a `RevisionDestsUpdate` it sends. In `func (rt *revisionThrottler) handleUpdate(...)`, looks like the function is also not expecting both `ClusterIPDest` and `Dests` to be set, because we are returning and not setting `rt.clusterIPDest` if `update.Dests != nil`. Could you explain this test case, and also is there a use case for both `ClusterIPDest` and `Dests` to be set?